RptToXml/Program.cs

```diff
@@ -12,44 +12,48 @@ static void Main(string[] args)
 		{
 			if (args.Length < 1)
 			{
-				Console.WriteLine("Usage: RptToXml.exe <RPT filename | wildcard> [outputfilename]");
+				Console.WriteLine("Usage: RptToXml.exe < -r | RPT filename | wildcard> [outputfilename]");
 				Console.WriteLine("       outputfilename argument is valid only with single filename in first argument");
 				return;
 			}
 
 			string rptPathArg = args[0];
-			bool wildCard = rptPathArg.Contains("*");
-			if (!wildCard && !ReportFilenameValid(rptPathArg))
-				return;
+			var rptPaths = new List<string>();
 
-			if (wildCard && args.Length > 1)
+			if ("-r".Equals(rptPathArg, StringComparison.InvariantCultureIgnoreCase))
 			{
-				Console.WriteLine("Output filename may not be specified with wildcard.");
-				return;
+				if (args.Length > 1)
+				{
+					Console.WriteLine("Output filename may not be specified with -r .");
+					return;
+				}
+				recursiveFileList(rptPaths, ".");
 			}
-
-			Trace.Listeners.Add(new TextWriterTraceListener(Console.Out));
-
-			var rptPaths = new List<string>();
-			if (!wildCard)
+			else if (rptPathArg.Contains("*"))
 			{
-				rptPaths.Add(rptPathArg);
+				if (args.Length > 1)
+				{
+					Console.WriteLine("Output filename may not be specified with wildcard.");
+					return;
+				}
+				var directory = Path.GetDirectoryName(rptPathArg);
+				if (String.IsNullOrEmpty(directory))
+				{
+					directory = ".";
+				}
+				var matchingFiles = Directory.GetFiles(directory, searchPattern: Path.GetFileName(rptPathArg));
+				rptPaths.AddRange(matchingFiles.Where(ReportFilenameValid));
+				if (rptPaths.Count == 0)
+				{
+					Trace.WriteLine("No reports matched the wildcard.");

```

Comment:
This should be console.writeline, not trace - we ahven't hooked up the trace listener yet.

Reply:
Or move the tracelistener init aabove rptPAths init as it was before